### DISCUSSION TITLE
FileDialog

### DISCUSSION THREAD

**Discussion by philippe06150** [gtk-rs/gtk4-rs#1464] [Q&A] [answered]
Hi, i'm a newbie in gtk-rs, that's my first post here, so I am reading the doc and testing examples posted on GITHUB

I downloaded 'textviewer' example and started to compile, so far so good till I got 

...
   Compiling gtk4 v0.8.0 (https://github.com/gtk-rs/gtk4-rs.git#239a0b18)
   Compiling gtk4_text_viewer v0.1.0 (C:\Users\phili\RUST\gtk4_text_viewer)
error[E0432]: unresolved import `gtk::FileDialog`
 --> src\main.rs:7:71
  |
7 | use gtk::{gio, glib, Application, ApplicationWindow, Builder, Button, FileDialog, TextView};
  |                                                                       ^^^^^^^^^^ no `FileDialog` in the root

OK i have checked in GTK doc, Filedialog is up since 4.10, so I thought something is obsolete
rustup update => version is 1.71.1 so the last one
in  toml i put 
[dependencies]
gtk = { git = "https://github.com/gtk-rs/gtk4-rs.git", package = "gtk4" }
so I am supposed to have the last issue

Where is my mistake ???

Thank u.

**Comment by bilelmoussaoui** [comment, marked as answer]
You have to enable the `v4_10` feature in order to be able to use these APIs. This is done to make it possible to use the bindings with old versions as well.
```toml
[dependencies]
gtk = { git = "https://github.com/gtk-rs/gtk4-rs.git", package = "gtk4" , features = ["v4_10"]}
```

**Comment by philippe06150** [comment]
"Hallelujah. You're my savior, man. My own personal Jesus Christ."

Never seen reference to this parameter 'features' before, now it works fine :-)

comprehensive answer in 13' is nice

Thanks

Nota : it is funny, i am under WIN11, the GTK window is not WIN style, but the Filedialog window is !! It is not a show stopper for what i want to do, but it is weird !

**Reply by bilelmoussaoui** [reply]
You can use some custom GTK theme that mimics the windows styling. Regarding the file chooser, that is expected because it just calls the platform API so under windows you get a windows file chooser.
